### DISCUSSION TITLE
Potential bug

### DISCUSSION THREAD

**Discussion by Lyxilion** [Rapptz/discord.py#6981] [General] [answered]
```python
@bot.event
async def on_raw_reaction_add(payload):
    if payload.member.id != 1234:  # working
        print('foo')

@bot.event
async def on_raw_reaction_remove(payload):
    if payload.member.id != 1234:  # not working
        print('bar')
```
With this code, if you remove a reaction, you raise on error : 
`AttributeError: 'NoneType' object has no attribute 'id'`

We found a fix, by using : 
`payload.user_id `
instead of :
`payload.member.id`

But the first code should be working from my understanding of the doc.
What do you think ? Am I missing something ?

**Comment by Rapptz** [comment, marked as answer]
[This is documented](https://discordpy.readthedocs.io/en/latest/api.html#discord.RawReactionActionEvent.member).

This is also a Discord limitation. They don't provide member information for reaction remove events for some reason. They claim performance reasons.

**Reply by Lyxilion** [reply]
Oh ok that was an oversight on my part then, thanks for the anwser ! 
Keep up the good job :)
